cypress/integration/eholdings/notes.spec.js

```diff
@@ -1,142 +1,129 @@
-        cy.get('#note-assign-button').click();
+      it('should redirect to note view page with edited details', () => {
+        cy.expect(Page.has({ url: matching(/\/eholdings\/notes\/[0-9a-z-]+/) }));
+        cy.expect(HTML(noteEdit).exists());
       });
+    });
+  });
 
-      it('should open note modal', () => {
-        cy.get('#notes-modal').then(div => expect(div).not.to.be.undefined);
-      });
+  describe('assigning a note', () => {
+    beforeEach(() => {
+      cy.do(Button('Assign / Unassign').click());
+    });
 
-      describe('searching for notes by name', () => {
-        before(() => {
-          cy.get('input[name="query"]').type('e2e');
-          cy.get('#notes-modal button[type="submit"]').click();
-        });
+    it('should open note modal', () => {
+      cy.expect(Modal('Assign / Unassign note').exists());
+    });
 
-        it('should show not empty result list', () => {
-          cy.get('#notes-modal-notes-list [class*="mclRowFormatterContainer--"]')
-            .then(rows => expect(rows).to.not.have.length(0));
-        });
+    describe('searching for notes by name', () => {
+      beforeEach(() => {
+        cy.do([
+          TextField('Note search').fillIn('e2e'),
+          Button('Search').click()
+        ]);
       });
 
-      describe('searching unassigned notes', () => {
-        before(() => {
-          cy.get('#clickable-filter-notesStatus-assigned').click();
+      it('should show not empty result list', () => {
+        cy.expect(Modal().find(MultiColumnList()).has({ rowCount: not(0) }));
+      });
+    });
 
-          cy.get('#notes-select-all-checkbox').click();
-          cy.get('#notes-modal-save-button').click();
-          cy.get('#note-assign-button').click();
+    describe('searching unassigned notes', () => {
+      beforeEach(() => {
+        cy.do([
+          Checkbox('Assigned').click(),
+          Checkbox('Assign / Unassign all notes').click(), // FIXME: This checkbox is disabled
+          Button('Save').click(),
+          Button('Assign / Unassign').click(),
+          Checkbox('Unassigned').click()
+        ]);
+      });
 
-          cy.get('#clickable-filter-notesStatus-unassigned').click();
-        });
+      // FIXME: This test should fail because notes can't be unassigned
+      it('should show not empty result list', () => {
+        cy.expect(Modal().find(MultiColumnList()).has({ rowCount: not(0) }));
+      });
 
-        it('should show not empty result list', () => {
-          cy.get('#notes-modal-notes-list [class*="mclRowFormatterContainer--"]')
-            .then(rows => expect(rows).to.not.have.length(0));
+      describe('assigning a note', () => {
+        let noteTitle;
+
+        beforeEach(() => {
+          cy.do([
+            Modal()
+              .find(MultiColumnListCell({ row: 0, columnIndex: 0 }))
+              .find(Checkbox('Assign / Unassign note')) // FIXME: This checkbox is disabled
```

Comment:
Please change the message so it says what should be done for the fix.

Reply:
Wrote comment for the whole describe block